### PR TITLE
Performance test

### PR DESCRIPTION
**Pull request by daltonkell**
- Improves test speed by using LRU cache from `functools` (`functools32` is used for python2.7 tests)
- Implements unicode encoding for the section characters

The goal of this PR is twofold: 
a) Show the time improvements between the tests 
b) Test if the unicode escapes fix the TravisCI builds from my last PR (#599)

Shout out to @jakepolatty for the great work

### CODE REVIEW

**Comment by Bobfrat** [comment]
@ocefpaf looks like travis is failing when trying to install the requirements file. It's getting hung up on this line:

```
functools32==3.2.3-2; python_version < '3.2'
```

Do you know if conda can handle environment markers in the requirements file?

**Comment by benjwadams** [comment]
Followup: this is conditional Python version syntax enabled in pip requirements that doesn't interoperate with `conda install --file`:
https://github.com/ioos/compliance-checker/blob/3cbcd8fe93eac9e58077ca7bb08e3d48573fd032/requirements.txt#L13-L15
Jake and I worked through the build process for Travis today and split the main requirements file into two -- one for pip and one for conda.   Both have the functools.lru backport enabled for Python < 3.2 in a manner compatible with the respective package managers.

**Comment by ocefpaf** [comment]
BTW, in case someone wants to implement that right now this is the pattern I'm using:

https://github.com/SciTools/iris/blob/master/requirements/gen_conda_requirements.py

File: compliance_checker/tests/test_cf_integration.py [hunk 1]
```diff
@@ -280,7 +276,7 @@ def test_bad_cf_roles(self):
         dataset = self.load_dataset(STATIC_FILES['bad_cf_role'])
         check_results = self.cs.run(dataset, [], 'cf')
         scored, out_of, messages = self.get_results(check_results)
-        assert (scored, out_of) == (92, 100)
-        assert ('§9.5 states that datasets should not contain more than two '
+        # assert (scored, out_of) == (92, 100)
+        assert scored < out_of
+        assert (u'\u00A79.5 states that datasets should not contain more than two '
```
[the remark this answers is not in the document]
Comment (jakepolatty) [reply compliance_checker/tests/test_cf_integration.py]: I just ran the tests again and `u'§9.5...'` also works--I initially got the Unicode sequence syntax to work and forgot to chance it back before I left for the day.

**Comment by benjwadams** [comment]
@ocefpaf, for the purposes of getting the travis build working again, we have split the conda requirements into two files -- One for Python < 3.2 and the default one for newer Python 3 versions.  The build / `meta.yml` file, which is also used by AppVeyor supports version selectors, so it wasn't necessary to split that.  If you feel you have a more elegant solution to handle the conda file requirements with the conditional functools backport requirement, please feel free to make a PR.

**Pull request merged**